pkg/apis/toolchain/v1alpha1/tiertemplate_types.go

```diff
@@ -0,0 +1,46 @@
+package v1alpha1
+
+import (
+	templatev1 "github.com/openshift/api/template/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// TierTemplateSpec defines the desired state of TierTemplate
+// +k8s:openapi-gen=true
+type TierTemplateSpec struct {
+	
+	// The type of the template. For example: "code", "dev" or "stage"
+	Type string `json:"type"`
+
+	// The revision of the corresponding template
+	Revision string `json:"revision"`
+
+	// Template the OpenShift Template to be used to provision a user's namespace or cluster-wide resources
+	Template templatev1.Template `json:"template"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// TierTemplate is the Schema for the tiertemplates API
+// +kubebuilder:resource:path=tiertemplates,scope=Namespaced
```

Comment:
What about adding `type` and `revision` columns?
```suggestion
// +kubebuilder:resource:path=tiertemplates,scope=Namespaced
// +kubebuilder:printcolumn:name="Type",type="string",JSONPath=`.spec.type`
// +kubebuilder:printcolumn:name="Revision",type="string",JSONPath=`.spec.revision`
```